static/app/views/performance/transactionSummary/transactionThresholdModal.tsx

```diff
@@ -0,0 +1,278 @@
+import * as React from 'react';
+import {Link} from 'react-router';
+import {css} from '@emotion/react';
+import styled from '@emotion/styled';
+import cloneDeep from 'lodash/cloneDeep';
+import set from 'lodash/set';
+
+import {addErrorMessage} from 'app/actionCreators/indicator';
+import {ModalRenderProps} from 'app/actionCreators/modal';
+import {Client} from 'app/api';
+import Button from 'app/components/button';
+import ButtonBar from 'app/components/buttonBar';
+import SelectControl from 'app/components/forms/selectControl';
+import {t, tct} from 'app/locale';
+import space from 'app/styles/space';
+import {Organization, Project} from 'app/types';
+import {defined} from 'app/utils';
+import EventView from 'app/utils/discover/eventView';
+import withApi from 'app/utils/withApi';
+import withProjects from 'app/utils/withProjects';
+import Input from 'app/views/settings/components/forms/controls/input';
+import Field from 'app/views/settings/components/forms/field';
+
+export enum TransactionThresholdMetric {
+  TRANSACTION_DURATION = 'duration',
+  LARGEST_CONTENTFUL_PAINT = 'lcp',
+}
+
+export const METRIC_CHOICES = [
+  {label: t('Transaction Duration'), value: 'duration'},
+  {label: t('Largest Contentful Paint'), value: 'lcp'},
+];
+
+type Props = {
+  api: Client;
+  organization: Organization;
+  transactionName: string;
+  onApply: (threshold, metric) => void;
+  projects: Project[];
+  eventView: EventView;
+  transactionThreshold: number | undefined;
+  transactionThresholdMetric: TransactionThresholdMetric | undefined;
+} & ModalRenderProps;
+
+type State = {
+  threshold: number | undefined;
+  metric: TransactionThresholdMetric | undefined;
+  error: string | null;
+};
+
+class TransactionThresholdModal extends React.Component<Props, State> {
+  state: State = {
+    threshold: this.props.transactionThreshold,
+    metric: this.props.transactionThresholdMetric,
+    error: null,
+  };
+
+  getProject() {
+    const {projects, eventView} = this.props;
+    const projectId = String(eventView.project[0]);
+    const project = projects.find(proj => proj.id === projectId);
+
+    return project;
+  }
+
+  handleApply = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const {api, closeModal, organization, transactionName} = this.props;
+
+    const project = this.getProject();
+    if (!defined(project)) {
+      return;
+    }
+
+    const transactionThresholdUrl = `/organizations/${organization.slug}/project-transaction-threshold-override/`;
+
+    api
+      .requestPromise(transactionThresholdUrl, {
+        method: 'POST',
+        includeAllArgs: true,
+        query: {
+          project: project.id,
+        },
+        data: {
+          transaction: transactionName,
+          threshold: this.state.threshold,
+          metric: this.state.metric,
+        },
+      })
+      .then(() => {
+        closeModal();
```

Comment:
So since you aren't doing `setState` here, you might have syncing issues between external and local data. 

The above code
```js
  state: State = {
    threshold: this.props.transactionThreshold
```

Will only run on the components first init. Have you tried re-opening the modal after an apply? I'm not 100% that our modal helpers perhaps deal with this, but it's something to be aware of.

If it is the case that these two values get out of sync, you can use one of the many functions to check incoming props and update state, or even easier, use the `key` trick to cause a new instance to get made when one of those props updates.

```js
<TransactionThresholdModal
key={transactionThreshold + transactionThresholdMetric}
```

Will cause the modal to be recreated any time the two values change externally. For something as simple as this form the cost is minimal to do so.

Reply:
Yeah the modal has the correct values when re-opened after apply or reset. I think "openModal" just initializes it again